presto-main/src/test/java/io/prestosql/sql/planner/assertions/ObjectMatcher.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.sql.planner.assertions;
+
+import java.util.Objects;
+import java.util.function.Predicate;
+
+import static java.util.Objects.requireNonNull;
+
+public class ObjectMatcher<T>
+{
+    private final Predicate<T> matcher;
```

Comment:
I would simply use `Predicate<T>` instead of `ObjectMatcher`. I don't see a value added here.